src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -190,7 +190,6 @@ class PeerDiscoveryManager(
 
     maybeDiscoveredNodes
       .map(_ ++ alreadyDiscoveredNodes)
-      .map(_.filterNot(isLocalNode))
```

Comment:
Why is it better to return the local node as a potential peer?

Reply:
I removed this because that filtering was happening twice

Reply:
Hm, I think it wasn't redundant, though, because [getting nodes from the discovery library](https://github.com/input-output-hk/mantis/blob/b81948c06cc1abaeaf6dc59c6eebfc8f56e8a22f/src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala#L186-L188) will have it. 

You can remove it, just be aware to skip it in the `PeerManagerActor`; maybe it's cheaper to filter it there.

Reply:
Although in that case it would make sense to remove it completely from this file.

Reply:
`alreadyDiscoveredNodes` also filters the localNode, that is why I removed it

Reply:
True, however `maybeDiscoveredNodes` can still have it. Perhaps you can modify that like so:
```scala
    val maybeDiscoveredNodes: Task[Set[Node]] =
      maybeDiscoveryService.fold(Task.pure(Set.empty[Node])) {
        _.getNodes.map { nodes =>
          nodes.map(toNode).filterNot(isLocalNode)
        }
      }
```